### ISSUE TITLE
False-positive unused variable: variable properties

### ISSUE THREAD

**Issue by AllenJB** [Slamdunk/phpstan-extensions#16]
```php
<?php

namespace AllenJB;

class Test
{

    function foo(array $numVals)
    {
        $arr = [
            "1" => "propOne",
            "2" => "propTwo",
        ];

        foreach ($numVals as $num => $v) {
            $key = $arr[$num];
            $this->{$key} = $v;
        }
    }
}
```

```
 ------ --------------------------------------------------------
  Line   Test.php
 ------ --------------------------------------------------------
  8      [Line  16] Function foo() has an unused variable $key.
 ------ --------------------------------------------------------
```

**Comment by Slamdunk** [comment]
Bugfix released in [3.5.2](https://github.com/Slamdunk/phpstan-extensions/releases/tag/v3.5.2)
